File: PickUp-Front/src/components/ReservationCustomer.js
import React from "react";
import { View, StyleSheet, TouchableOpacity, Image } from "react-native";
import Swipeable from "react-native-gesture-handler/Swipeable";
import { MaterialCommunityIcons } from "@expo/vector-icons";

import colors from "../config/colors";
import AppText from "./AppText";

function ReservationCustomer({
  backgroundColor = colors.secondary,
  item,
  onPress,
  renderRightActions,
}) {
  return (
    <Swipeable renderRightActions={renderRightActions}>
      <TouchableOpacity activeOpacity={0.6} onPress={onPress}>
        <View style={[styles.container, { backgroundColor }]}>
          <Image style={styles.image} source={{ uri: item.url }} />
          <View style={styles.details}>
            <AppText style={styles.title}>{item.name}</AppText>
            <AppText>{item.resDate}</AppText>
            <AppText>
              {item.startHour.slice(0, 5)} -{item.endHour.slice(0, 5)}
            </AppText>
            <AppText>Durée : {item.startHour - item.endHour}</AppText>
            <AppText>Nombre de Personne: {item.numPlaceAvailable}</AppText>
          </View>
          <MaterialCommunityIcons
            color={colors.medium}
            name="chevron-right"
            size={25}
          />
        </View>
      </TouchableOpacity>
    </Swipeable>
  );
}

const styles = StyleSheet.create({
  container: {
    borderRadius: 35,
    overflow: "hidden",
    padding: 10,
    paddingLeft: 20,
    margin: 5,
    flexDirection: "row",
    alignItems: "center",
  },
  details: {
    marginLeft: 15,
  },
  imageContainer: {},
  image: {
    width: 100,
    height: "100%",
    borderRadius: 15,
  },
  title: {
    fontSize: 20,
    fontWeight: "bold",
  },
});

export default ReservationCustomer;

File: PickUp-Back/PickUp.DAL/Models/ReservationCustomer.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace PickUp.DAL.Models
{
    public class ReservationCustomer
    {
        public int ReservationId { get; set; }
        public int UserId { get; set; }
        public string Name { get; set; }
        public string ResDate { get; set; }
        public string StartHour { get; set; }
        public string EndHour { get; set; }
        public string Url { get; set; }
        public int NumPlaceAvailable { get; set; }

        
        public ReservationCustomer(int userId, string name,
            string resDate, string startHour, string endHour,
            int numPlaceAvailable, string url)
        {
            UserId = userId;
            Name = name;
            ResDate = resDate;
            StartHour = startHour;
            EndHour = endHour;
            NumPlaceAvailable = numPlaceAvailable;
            Url = url;
        }

        internal ReservationCustomer(int reservationId, int userId,
            string name, string resDate, string startHour,
            string endHour, int numPlaceAvailable, string url):
            this(userId, name, resDate, startHour, endHour, numPlaceAvailable, url)
        {
            ReservationId = reservationId;
        }
        public ReservationCustomer(int reservationId, int userId, string resDate,
            string startHour, string endHour, int numPlaceAvailable)
        {
            ReservationId = reservationId;
            UserId = userId;
            ResDate = resDate;
            StartHour = startHour;
            EndHour = endHour;
            NumPlaceAvailable = numPlaceAvailable;
        }
        public ReservationCustomer(string resDate,
            string startHour, string endHour, int numPlaceAvailable, int userId)
        {
            UserId = userId;
            ResDate = resDate;
            StartHour = startHour;
            EndHour = endHour;
            NumPlaceAvailable = numPlaceAvailable;
        }

    }
}
